code/game/atoms.dm

```diff
@@ -726,22 +726,26 @@
 	SEND_SIGNAL(src, COMSIG_ATOM_UPDATE_OVERLAYS, .)
 
 /// Checks if the colors given are different and if so causes a greyscale icon update
-/atom/proc/set_greyscale_colors(list/colors)
+/// The colors argument can be either a list or the full color string
+/atom/proc/set_greyscale_colors(list/colors, update=TRUE)
 	SHOULD_CALL_PARENT(TRUE)
-	var/new_colors = colors.Join("")
-	if(greyscale_colors == new_colors)
+	if(istype(colors))
+		colors = colors.Join("")
+	if(greyscale_colors == colors)
 		return
-	greyscale_colors = new_colors
+	greyscale_colors = colors
 	if(!greyscale_config)
 		return
-	update_greyscale()
+	if(update)
+		update_greyscale()
 
 /// Checks if the greyscale config given is different and if so causes a greyscale icon update
-/atom/proc/set_greyscale_config(new_config)
+/atom/proc/set_greyscale_config(new_config, update=TRUE)
```

Comment:
I don't like bool parameters like this, why should consumers of this not just set `greyscale_config` directly?